mlflow/sklearn/__init__.py

```diff
@@ -455,6 +461,311 @@ def load_model(model_uri):
+
+    def get_run_id_for_model(self, model):
+        if hasattr(model, '_mlflow_uuid'):
+            return self._model_id_to_run_id_map.get(model._mlflow_uuid, None)
+        return None
+
+    @staticmethod
+    def get_model_id(model):
+        return model._mlflow_uuid
+
+    @staticmethod
+    def is_metrics_value_loggable(metric_value):
+        return np.isscalar(metric_value)
+
+    def register_model(self, model, run_id):
+        """
+        In `patched_fit`, we need register the model with the run_id used in `patched_fit`
+        So that in following metric autologging, the metric will be logged into the registered
+        run_id
+        """
+        model._mlflow_uuid = uuid4()
+        self._model_id_to_run_id_map[model._mlflow_uuid] = run_id
+
+    @staticmethod
+    def gen_name_with_index(name, index):
+        assert index >= 0
+        if index == 0:
+            return name
+        else:
+            # Use '-' as the separator between name and index,
+            # The '-' is not valid character in python var name
+            # so it can prevent name conflicts after appending index.
+            return f'{name}-{index + 1}'
+
+    @staticmethod
+    def gen_dataset_sample_rows(eval_dataset):
+        n_sample = max(len(eval_dataset), INPUT_EXAMPLE_SAMPLE_ROWS)
+        return eval_dataset[:n_sample]
+
+    def register_eval_dataset(self, model, eval_dataset):
+        """
+        Register eval dataset into eval_dataset_info_map, it will do:
+         1. inspect eval dataset var name.
+         2. check whether eval_dataset_info_map already registered this eval dataset.
+            will check by object id and comparing row samples.
+            Note: only check object id is not sufficient, id may be reused.
+         3. register eval dataset (with id and row samples information)
+         4. return eval dataset name with index.
+        """
+        eval_dataset_name = _inspect_original_var_name(eval_dataset)
+        eval_dataset_id = id(eval_dataset)
+        if eval_dataset_name is None:
+            eval_dataset_name = 'unknown_dataset'
+
+        model_id = self.get_model_id(model)
+        registered_dataset_list = self._eval_dataset_info_map[model_id][eval_dataset_name]
+
+        row_samples = self.gen_dataset_sample_rows(eval_dataset)
+        index = len(registered_dataset_list)
+        for i, (id_i, row_samples_i) in enumerate(registered_dataset_list):
+            if eval_dataset_id == id_i and np.array_equal(row_samples, row_samples_i):
+                index = i
+                break
+
+        if index == len(registered_dataset_list):
+            # register new eval dataset
+            registered_dataset_list.append((eval_dataset_id, row_samples))
+            self._eval_dataset_row_samples_artifact_need_update[model_id] = True
+
+        return self.gen_name_with_index(eval_dataset_name, index)
+
+    def register_prediction_result(self, model, eval_dataset_name, predict_result):
```

Comment:
Just to confirm, in the following code, `accuracy_score` logs nothing right?

```python
import xgboost as xgb
import sklearn.metrics

model = xgb.train()
preds = model.predict(...)
acc = sklearn.metrics.accuracy_score(...)  # this is not logged
```

Reply:
for xgb we do not patch xgb train/predict, so it should not work

We only patch estimtor class in sklearn lib.

Reply:
Got it, can we indicate this in the doc?